URLs with :: are OK and should not raise assertion errors

For conda-build's perl skeleton generator, we can end up with URLs like:

http://api.metacpan.org/v0/module/Test::More

Unfortunately, conda prevents us from actually using those URLs:

```
  File "/Users/msarahan/miniconda2/lib/python2.7/site-packages/conda/fetch.py", line 354, in download
    assert "::" not in str(url), url
AssertionError: http://api.metacpan.org/v0/module/Test::More
```

Please partially revert https://github.com/conda/conda/commit/39605e01ccd05b5af5ebeceeacaafe652f4b32e4

--- a/conda/fetch.py
+++ b/conda/fetch.py
@@ -351,7 +351,6 @@ def fetch_pkg(info, dst_dir=None, session=None):
 
 
 def download(url, dst_path, session=None, md5=None, urlstxt=False, retries=None):
-    assert "::" not in str(url), url
     assert "::" not in str(dst_path), str(dst_path)
     if not offline_keep(url):
         raise RuntimeError("Cannot download in offline mode: %s" % (url,))

